samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c

```diff
@@ -679,6 +650,64 @@ static void gen_delta_set(struct bt_mesh_model *model,
 	}
 }
 
+static void gen_level_move_get(struct bt_mesh_model *model,
+			       struct bt_mesh_msg_ctx *ctx,
+			       struct net_buf_simple *buf)
+{
+	struct net_buf_simple *msg = NET_BUF_SIMPLE(2 + 5 + 4);
+	struct generic_level_state *state = model->user_data;
+
+	bt_mesh_model_msg_init(msg, BT_MESH_MODEL_OP_GEN_LEVEL_STATUS);
+	net_buf_simple_add_le16(msg, state->level);
+
+	if (state->transition->counter) {
+
+		if (state->last_delta < 0) {
+			net_buf_simple_add_le16(msg, INT16_MIN);
+		} else if (state->last_delta > 0) {
+			net_buf_simple_add_le16(msg, INT16_MAX);
+		}
```

Comment:
What if last_delta is 0? I that case you'd create a corrupted message without a Target Level field.

Reply:
when client sends delta = 0 means stop the transition.
Plus it will also set state->transition->counter = 0

That means following lines would not execute in that conditions

```
if (state->transition->counter) {
		if (state->last_delta < 0) {
			net_buf_simple_add_le16(msg, INT16_MIN);
		} else if (state->last_delta > 0) {
			net_buf_simple_add_le16(msg, INT16_MAX);
		}

		/* This is as per PTS requirement */
		net_buf_simple_add_u8(msg, UNKNOWN_VALUE);
}

```

Reply:
Done !!

Reply:
done !!